File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/user/exceptions/FailLoginException.java
package com.hgianastasio.biblioulbrav2.core.user.exceptions;

/**
 * Created by heitorgianastasio on 4/26/17.
 */

public class FailLoginException extends Exception {
    String cgu;

    public FailLoginException(String cgu) {
        this.cgu = cgu;
    }
    public FailLoginException(String cgu, Throwable throwable) {
        super(throwable);
        this.cgu = cgu;
    }

    @Override
    public String getMessage() {
        return String.format("O cgu %s é inválido",cgu);
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/user/exceptions/FailLoginException.kt
package com.hgianastasio.biblioulbrav2.core.user.exceptions

/**
 * Created by heitorgianastasio on 4/26/17.
 */
class FailLoginException : Exception {
    var cgu: String?

    constructor(cgu: String?) {
        this.cgu = cgu
    }

    constructor(cgu: String?, throwable: Throwable?) : super(throwable) {
        this.cgu = cgu
    }

    override val message: String
        get() = String.format("O cgu %s é inválido", cgu)
}